src/launcher_gui.js

```diff
@@ -21,89 +26,123 @@ app.on('second-instance', () => {
 });
 
 app.prependListener('ready', () => {
-	const { wizard } = require('./launcher_wizard');
-
-	const display = electron.screen.getPrimaryDisplay();
-	const sWidth = display.workAreaSize.width;
-	const width = 800;
-	const height = process.platform === 'win32' ? 418 : 380 + 8;
-
-	let windowOpts = {
-		x: (sWidth - width) / 2,
-		// y: (sHeight - height) / 2,
-		y: 100,
-		width: width,
-		height: height,
-		show: false,
-		icon: `${__dirname}/renderer/spring-icon.png`,
-		webPreferences: {
-			nodeIntegration: true,
-			enableRemoteModule: true,
-		},
-	};
-	windowOpts.resizable = false;
-	Menu.setApplicationMenu(null);
-	mainWindow = new BrowserWindow(windowOpts);
-
-	mainWindow.loadFile(`${__dirname}/renderer/index.html`);
-	if (isDev) {
-		// mainWindow.webContents.openDevTools();
-	}
+	// TODO: is there a better way to get yargs' positional args ? Did not find
+	// it ...
+	if (argv._[0]) {
+		// If a replay file was passed as argument to spring-launcher, then we
+		// don't open the GUI, we just start the replay.
+
+		// TODO: when we support downloading assets (map/game/engine) we might
+		// need to show the GUI anyway, to provide feedback on the
+		// downloads/ask for user confirmation.
+		const { launcher } = require('./engine_launcher');
+
+		const replayPath = decodeURI(url.parse(argv._[0]).pathname);
```

Comment:
I think you should first check if the replay exists, otherwise users get "Engine not found" error when invalid replay path is specified

Reply:
Did you check that ? From my own testing, this will launch the engine, which will then show a message "demofile not found: <path to the replay>".

I guess I can also check that it exists here, so we don't have to spawn the engine at all

Reply:
- From my own testing, this will launch the engine, which will then show a message "demofile not found: ".
Ah I was giving in truly bogus names, something that cannot even be parsed as an engine
- I guess I can also check that it exists here, so we don't have to spawn the engine at all
Yeah I think it'd be best to do as many checks as we can to avoid trying to launch invalid files.

Reply:
Yeah I added the check!